ztp/siteconfig-generator/siteConfig/filesHandler.go

```diff
@@ -0,0 +1,67 @@
+package siteConfig
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+func GetFiles(path string) ([]os.FileInfo, error) {
+	fileInfo, err := os.Stat(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if fileInfo.IsDir() {
+		return ioutil.ReadDir(path)
+	}
+
+	return []os.FileInfo{fileInfo}, nil
+}
+
+func ReadFile(filePath string) ([]byte, error) {
+	return ioutil.ReadFile(filePath)
+}
+
+func ReadResourceFile(fileName string) ([]byte, error) {
+	var dir = ""
+	var err error = nil
+	var ret []byte
+
+	ex, err := os.Executable()
+	if err != nil {
+		return nil, err
+	}
+	dir = filepath.Dir(ex)
+	ret, err = ReadFile(dir + "/" + localExtraManifestPath + "/" + fileName)
+
+	// added fail safe for test runs as `os.Executable()` will fail for tests
+	if err != nil {
+
+		dir, err = os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		ret, err = ReadFile(dir + "/" + localExtraManifestPath + "/" + fileName)
+	}
+	return ret, err
+}
+
+func GetResourceFiles() ([]os.FileInfo, string, error) {
+	ex, err := os.Executable()
+	if err != nil {
+		return nil, "", err
+	}
+
+	dir := filepath.Dir(ex)
+	files, err := GetFiles(dir + "/" + localExtraManifestPath)
+	if err != nil {
+		dir, err = os.Getwd()
+		if err != nil {
+			return nil, "", err
+		}
+		files, err = GetFiles(dir + "/" + localExtraManifestPath)
+	}
+	return files, dir, err
```

Comment:
Here - We're returning files that are relative to `dir + "/" + localExtraManifestPath` so why not just
```
    return files, dir + "/" + localExtraManifestPath, err
```
Then the caller doesn't have to know where it found them.

Reply:
okay, got it will do. just get lost while cleaning the code